src/resourcepacks/ResourcePackManager.php

```diff
@@ -106,6 +111,11 @@ public function __construct(string $path, \Logger $logger){
 				if($newPack instanceof ResourcePack){
 					$this->resourcePacks[] = $newPack;
 					$this->uuidList[strtolower($newPack->getPackId())] = $newPack;
+
+					$keyPath = Path::join($this->path, $pack . ".key");
+					if(file_exists($keyPath)){
+						$this->encryptionKeys[strtolower($newPack->getPackId())] = Utils::assumeNotFalse(file_get_contents($keyPath));
```

Comment:
This will crash if the file permissions are incorrect. `ErrorToExceptionHandler::trapAndRemoveFalse()` should be used instead. Catch the `ErrorException` it throws and rethrow it as a `ResourcePackException`.